pilz_control/test/unittest_pilz_joint_trajectory_controller.cpp

```diff
@@ -482,35 +494,37 @@ TEST_P(PilzJointTrajectoryControllerIsExecutingTest, testSingleActionGoal)
   progressInTime(goal_duration + default_update_period);
   updateController();
 
-  EXPECT_TRUE(invokeIsExecuting(is_executing_result));
-  EXPECT_FALSE(is_executing_result) << "Controller is executing unexpectedly";
+  EXPECT_FALSE(isControllerExecuting()) << "Controller is executing unexpectedly";
 }
 
+using pilz_testutils::IsWarn;
+using ::testing::_;
 /**
- * @brief Test if the controller starts and completes the execution of a single trajectory command.
+ * @brief Tests that the controller does not execute if a trajectory is sent via command interface.
  */
 TEST_P(PilzJointTrajectoryControllerIsExecutingTest, testSingleCommandMessage)
 {
   ASSERT_TRUE(performFullControllerStartup());
 
-  /**********
-   * Step 1 *
-   **********/
   ros::Duration goal_duration{ 2.0 };
   GoalType goal{ generateSimpleGoal(goal_duration) };
 
+  log4cxx::LoggerPtr logger = log4cxx::Logger::getLogger(ROSCONSOLE_ROOT_LOGGER_NAME);
+  auto mock_logger = new pilz_testutils::LoggerMock();
+  logger->addAppender(mock_logger);
+
+  EXPECT_CALL(*mock_logger,
+              append(IsWarn("For safety reasons the trajectory command interface is deactivated "
+                            "(for more information see https://github.com/ros-controls/ros_controllers/issues/493). "
+                            "Please use the action interface instead."),
+                     _))
+      .Times(1);
+
   trajectory_command_publisher_->publish(goal.trajectory);
-  bool is_executing_result;
-  EXPECT_TRUE(waitForIsExecutingResult(true, is_executing_result));
 
-  /**********
-   * Step 2 *
-   **********/
-  ros::Duration default_update_period{ DEFAULT_UPDATE_PERIOD_SEC };
-  progressInTime(goal_duration + default_update_period);
-  updateController();
-  EXPECT_TRUE(invokeIsExecuting(is_executing_result));
-  EXPECT_FALSE(is_executing_result) << "Controller is executing unexpectedly";
+  EXPECT_TRUE(noExecutionObserving(goal_duration + ros::Duration(DEFAULT_UPDATE_PERIOD_SEC)));
+
+  logger->removeAppender(mock_logger);
```

Comment:
Can you be sure, that the logging message appeared, when the test completes? Just checking, because I'm not very familiar with the loggers.

Reply:
Imho yes because as far as is understand this there is no asynchronous behavior.

Reply:
Done.